File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictionary = {}
        for i in range(len(nums)):
            difference = target - nums[i]
            if difference not in dictionary.keys():
                dictionary[nums[i]] = i
            else:
                return [dictionary[difference], i]
                

# @lc code=end

File: code/1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
import java.util.*;
import java.io.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        int[] ret = new int[2];
        for (int i = 0;i < nums.length;i++) {
            if (hash.containsKey(nums[i])) {
                ret[0] = hash.get(nums[i]);
                ret[1] = i;
                return ret;
            }
            else {
                hash.put(target-nums[i], i);
            }
        }
        return ret;
    }
}

// @lc code=end

